genesyscloud/architect_user_prompt/data_source_genesyscloud_architect_user_prompt.go

```diff
@@ -0,0 +1,38 @@
+package architect_user_prompt
+
+import (
+	"context"
+	"fmt"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceUserPromptRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	sdkConfig := m.(*provider.ProviderMeta).ClientConfig
+	proxy := getArchitectUserPromptProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+	nameArr := []string{name}
```

Comment:
Best to change it, there is no need to create an array here when we can just use a string

Reply:
why does the SDK have an array input?  it would be easier to keep it as closer to the SDK requirements for future changes

Reply:
The sdk has an array input because that is what's defined in the api's swagger. I assume it is if you want to find objects using more than one name so basically "return all objects with name1 or name2 etc". This isn't functionality we would need in cx as code since we are only trying to find one object based on one name. It's better if we keep this function inline with everything else and just pass a single name since that functionality should never change even if the api does

Reply:
again based on assumptions, keeping it closer to the orginial required type doesn't need an explanation at all.

Reply:
It's not an assumption. Sending in multiple names would mean we are trying to find an object in genesycloud using multiple names. And object can only have one name